CRM/Event/Form/ManageEvent/TabHeader.php

```diff
@@ -199,9 +199,11 @@ public static function process(&$form) {
           $action = 'browse';
         }
 
-        $tabs[$key]['link'] = CRM_Utils_System::url("civicrm/event/manage/{$key}",
-          "{$reset}action={$action}&id={$eventID}&component=event{$tabs[$key]['qfKey']}"
-        );
+        $link = "civicrm/event/manage/{$key}";
+        $query = "{$reset}action={$action}&id={$eventID}&component=event{$tabs[$key]['qfKey']}";
+
+        $tabs[$key]['link'] = (isset($value['link']) ? self::url($value['link']) :
+          CRM_Utils_System::url($link, $query));
```

Comment:
@totten, it didn't occur to me that the link might be to a CMS page. Turns out that setting the snippet is unnecessary -- if the Angular page is being called from a tab or a modal, the snippet parameter is already set -- so we can do without CRM_Utils_System::snippetUrl(...) altogether.

Perhaps I was engaging in too much extension-developer hand-holding.